luaPlugin/luaP.cpp

```diff
@@ -1379,56 +1590,185 @@ sol::state* luaP::init(std::string& luaFilePath, std::string& modPath)
+	types.building.set("buildingData", &building::bData);
+	types.building.set("level", &building::level);
+	types.building.set("hp", &building::hp);
+	types.building.set("settlement", &building::settlement);
+	/***
+	Get name of building type (chain)
+
+	@function building:getType
+	@treturn string buildingType (building chain name)
 	@usage
-	ourGuildStanding=settlementStruct:getGuildStanding(0);
+	if building:getType() == "core_building" then
+		--something
+	end
 	*/
-	types.settlementStruct.set_function("getGuildStanding", &settlementHelpers::getGuildStanding);
+	types.building.set_function("getType", &buildingStructHelpers::getType);
 	/***
-	Set settlement standing points with guild
-	@function settlementStruct:setGuildStanding
-	@tparam int guild_id
-	@tparam int standing
+	Get name of building level
+
+	@function building:getName
+	@treturn string buildingName
 	@usage
-	settlementStruct:setGuildStanding(0,300);
+	if building:getName() == "large_stone_wall" then
+		--something
+	end
 	*/
-	types.settlementStruct.set_function("setGuildStanding", &settlementHelpers::setGuildStanding);
+	types.building.set_function("getName", &buildingStructHelpers::getName);
+
+
+	///BuildingsQueue table section
+	//@section buildingsQueueTable
+
+	/***
+	Basic buildingsQueue table
+
+	@tfield int firstIndex index (0-5) of building in first position (currently under construction), not necessarily 0 as buildingsQueue uses a circular buffer
+	@tfield int lastIndex index (0-5) of last building in the queue
+	@tfield int currentBuildingPosition position in the queue (1-6) that is currently under construction, usually 1
+	@tfield int numBuildingsInQueue maximum is 6
+	@tfield buildingInQueue[6] buildingInQueue index+1 (1-6), see usage for getQueueName()
+
+	@table buildingsQueue
+	*/
+	types.buildingsQueue = luaState.new_usertype<buildingsQueue>("buildingsQueue");
+	types.buildingsQueue.set("firstIndex", &buildingsQueue::firstIndex);
+	types.buildingsQueue.set("lastIndex", &buildingsQueue::lastIndex);
+	types.buildingsQueue.set("currentBuildingPosition", &buildingsQueue::currentBuildingIndex);
+	types.buildingsQueue.set("numBuildingsInQueue", &buildingsQueue::buildingsInQueue);
+	types.buildingsQueue.set("buildingInQueue", sol::property([](buildingsQueue& self) { return std::ref(self.items); }));
+
+
+	///BuildingInQueue table section
+	//@section buildingInQueueTable
+
+	/***
+	Basic buildingInQueue table
+
+	@tfield building building nil if building doesn't exist yet
+	@tfield settlementStruct settlement
+	@tfield int currentLevel
+	@tfield int previousLevel
+	@tfield int buildCost
+	@tfield int buildTurnsPassed
+	@tfield int buildTurnsRemaining
+	@tfield int percentBuilt
+	@tfield getQueueType getQueueType
+	@tfield getQueueName getQueueName
+
+	@table buildingInQueue
+	*/
+	types.buildingInQueue = luaState.new_usertype<buildingInQueue>("buildingInQueue");
+	types.buildingInQueue.set("buildingData", &buildingInQueue::buildingData);
```

Comment:
need record in descriotion(i.e. @tfield buildingData etc

Reply:
The building_data structure is also present in the building table section without documentation. I figured it was somehow needed for the name getters to work so I just copied it over for the buildingInQueue table.

The building_data structure isn't set up in Lua. Aside from accessing 'isDestroyAllowed' (which is a string for some reason?), I don't see the point of adding this structure. If it's not needed, it should probably just be removed from building and buildingInQueue.